### PR TITLE
chore(slack): log full response for rule.fail.slack_post

### PR DESCRIPTION
**Pull request by cathteng**
Sometimes the `error` field in `ApiError` in an empty string. If we log the full response we can see what is happening when we don't get an `error` field back in the response.

### CODE REVIEW

File: src/sentry/integrations/slack/client.py [hunk 1]
```diff
@@ -137,5 +137,7 @@ def request(
             **kwargs,
         )
         if not raw_response and not response.json.get("ok"):
+            if kwargs.get("log_response_with_error", False):
+                self.logger.info("rule.fail.slack_post.log_response", extra={"response": response})
```
Comment (ykamo001) [inline src/sentry/integrations/slack/client.py:141]: the response is a django class object no? Won't this just log the memory address and not the contents? Are we looking for `response.__dict__`?

**Review by ykamo001 — APPROVED** [review]

File: src/sentry/integrations/slack/actions/notification.py [hunk 2]
```diff
@@ -109,7 +109,9 @@ def send_notification(event: GroupEvent, futures: Sequence[RuleFuture]) -> None:
 
             client = SlackClient(integration_id=integration.id)
             try:
-                client.post("/chat.postMessage", data=payload, timeout=5)
+                client.post(
```
Comment (ykamo001) [inline src/sentry/integrations/slack/actions/notification.py:112]: should we add this to metric alerts flow as well?

Comment (cathteng) [reply src/sentry/integrations/slack/actions/notification.py:112]: does the metric alert message use blocks?

Comment (cathteng) [reply src/sentry/integrations/slack/actions/notification.py:112]: i actually don't think it uses block kit, looking at the class

Comment (cathteng) [reply src/sentry/integrations/slack/actions/notification.py:112]: i'm okay isolating to just issue alerts for now

**Pull request merged**